src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -2814,6 +2819,54 @@ public void testSideEffect() {
     tracer.setExpected("sideEffect", "sleep PT1S", "executeActivity customActivity1");
   }
 
+  private static final Map<String, Queue<Long>> mutableSideEffectValue =
+      Collections.synchronizedMap(new HashMap<>());
+
+  public static class TestMutableSideEffectWorkflowImpl implements TestWorkflow1 {
+
+    @Override
+    public String execute(String taskList) {
+      StringBuilder result = new StringBuilder();
+      for (int i = 0; i < 4; i++) {
+        long value =
+            Workflow.mutableSideEffect(
+                "id1",
+                Long.class,
+                (o, n) -> n > o,
+                () -> mutableSideEffectValue.get(taskList).poll());
+        if (result.length() > 0) {
+          result.append(", ");
+        }
+        result.append(value);
+        if (i >= 3) {
+          Workflow.sleep(Duration.ofSeconds(1));
```

Comment:
Can you add a comment to say that this sleep is to test the replay mode logic? It's not obvious from the code.